src/Compilers/CSharp/Portable/Symbols/Source/ParameterHelpers.cs

```diff
@@ -98,6 +109,33 @@ internal static class ParameterHelpers
             return parameters;
         }
 
+        private static bool IsPropertyWithoutBody(Symbol owner)
+        {
+            if (owner is SourcePropertySymbol)
```

Comment:
Does it have to be a SourcePropertySymbol or can we cover all PropertySymbols? Also, in compiler code, we ,usually, do Kind check instead of type check whenever possible. If Kind check is not possible, we avoid doing type check twice by storing result of an ```as Type``` operation into a local and then checking it for null. Note, a check for null for symbols should be done with custing them to ```object``` first.